### DISCUSSION TITLE
Invitation mail - registration

### DISCUSSION THREAD

**Discussion by filipagh** [fief-dev/fief#146] [Q&A] [answered]
hello, is there option to do not allow open registration but have ability to send email with invitation which will lead to registration form ?

like allow registration only with invitation token and without valid invitation refuse registration (show invalid token )

**Comment by fief-bailiff** [comment]
Hail, @filipagh 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment, marked as answer]
Hi @filipagh 👋 

Currently, there is no invite-only feature, sorry. This is something we have on the roadmap.

A workaround could be to implement the token verification on your side: the user would be allowed to register (and login) on your Fief instance no matter what, but then your backend would require them to be verified with a token before being granted access to the app. Not ideal, but it should work :)

**Reply by filipagh** [reply]
i do not see it on roadmap https://github.com/orgs/fief-dev/projects/1/views/4
closest is valid email verivication but thats not same

**Reply by frankie567** [reply]
Added:

* https://github.com/orgs/fief-dev/projects/1/views/4?pane=issue&itemId=21517132
* https://github.com/orgs/fief-dev/projects/1/views/4?pane=issue&itemId=21517406

**Comment by filipagh** [comment]
for now i will disable registration on tenant so login page will not contain registration option 
and i will implement registration form in my app and use admin api key to register user if valid token provided

**Reply by frankie567** [reply]
That's also a perfectly valid solution 😄 Feel free to ask if you need help!

**Reply by shirkan** [reply]
> for now i will disable registration on tenant so login page will not contain registration option and i will implement registration form in my app and use admin api key to register user if valid token provided

@filipagh 
Can you please elaborate more how are you going to do this? 
I'm looking to do a similar thing (allow registration with a verification token).
Thanks

**Reply by filipagh** [reply]
@shirkan hello, now i have FE app and BE fastapi app, so i will add endpoint to BE which allow registration (email, password, register token)
admin in FE can generate registration mail and send it to some email with link to my registration form page here new user set email and password and send it to my BE (with registration token)

if registration is valid then i will use Fief admin [api key ](https://github.com/orgs/fief-dev/discussions/146#discussioncomment-5179758) and create new user with fief admin api https://\<acc\>.fief.dev/admin/api/docs  
like here https://github.com/orgs/fief-dev/discussions/70

**Reply by shirkan** [reply]
> @shirkan hello, now i have FE app and BE fastapi app, so i will add endpoint to BE which allow registration (email, password, register token) admin in FE can generate registration mail and send it to some email with link to my registration form page here new user set email and password and send it to my BE (with registration token)
> 
> if registration is valid then i will use Fief admin [api key ](https://github.com/orgs/fief-dev/discussions/146#discussioncomment-5179758) and create new user with fief admin api https://<acc>.fief.dev/admin/api/docs like here https://github.com/orgs/fief-dev/discussions/70

Got it! thanks a lot for sharing, gl
